pkg/test/kube/accessor.go

```diff
@@ -248,7 +256,7 @@ func (a *Accessor) GetService(ns string, name string) (*v12.Service, error) {
 }
 
 // CreateNamespace with the given name. Also adds an "istio-testing" annotation.
-func (a *Accessor) CreateNamespace(ns string, istioTestingAnnotation string) error {
+func (a *Accessor) CreateNamespace(ns string, istioTestingAnnotation string, injectionEnabled bool) error {
```

Comment:
Should we be doing this here? Here is my worry:
I know the Pilot Apps need injection. But the Policy Backend should not be getting injected as Mixer makes a direct call (as part of infrastructure layer).  We need to make sure that we have the right default/exception flags are set in the appropriate places.

I think it makes sense to make test namespace injection to be default. That is where the "theoretical application stuff that is being tested" should be put in. But I was thinking of the dependency namespace as a place where we put mocks/fakes pertaining to the control plane.

In that sense, *conceptually*, it makes sense to deploy the apps to the test namespace. Now, for practical purposes it makes a lot of sense to deploy it to dep. But, I'd take that as an exception, rather than the rule.

So, I propose making the system & dep namespaces to be non-injected (by default) and test namespace injected by default. App deployments goes to the dep namespace, but gets explicit labels in the deployments to get injected. 

WDYT?

Reply:
The Policy backend already excludes itself properly from injection.  I don't think there's a worry here. Most folks are going to enable automatic injection by default for all namespaces.  We should try to model a typical environment to the extent possible.

To be honest, I'm not sure I totally appreciate the distinction between the dependency and test namespaces. Currently I'm not even sure if the test namespace is even used. I suspect all of this will evolve in the near future.